src/utils.rs

```diff
@@ -164,16 +181,22 @@ pub enum Webxdc {
 }
 
 impl Webxdc {
-    pub fn get_str_path(&self) -> &'static str {
-        match self {
+    pub fn get_path(&self) -> Result<PathBuf> {
+        let filename = match self {
             Webxdc::Shop => STORE_XDC,
             Webxdc::Submit => SUBMIT_HELPER_XDC,
             Webxdc::Review => REVIEW_HELPER_XDC,
-        }
+        };
+        let path = assets_path()?.join(filename);
+        Ok(path)
     }
 
-    pub fn get_path(&self) -> PathBuf {
-        PathBuf::from(self.get_str_path())
+    pub fn get_str_path(&self) -> Result<String> {
+        let path = self.get_path()?;
+        let s = path
+            .to_str()
+            .with_context(|| format!("cannot convert path {} to string", path.display()))?;
+        Ok(s.to_string())
     }
```

Comment:
```suggestion
        self.get_path()?
            .to_str()
            .with_context(|| format!("cannot convert path {} to string", path.display()))
            .map(|str| str.to_string())
    }
```
For me this way nice, but that of cause is personal preference

Reply:
Might even be possible to further simplify the map with something like `to_owned()`